docs/pages/pmd/devdocs/pmd_devdocs_metrics_index.md

```diff
@@ -0,0 +1,107 @@
+---
+title:  Index of code metrics
+tags: [customizing]
+summary: "Index of the code metrics available out of the box to Java rule developers."
+last_updated: July 20, 2017
+sidebar: pmd_sidebar
+permalink: pmd_devdocs_metrics_index.html
+folder: pmd/devdocs
+toc:
+  minimumHeaders: 8
+---
+# Index of code metrics
+
+## Access to Foreign Data (ATFD)
+
+*Operation metric, class metric.*
+
+### Description
+
+
+## Cyclomatic Complexity (CYCLO)
+
+*Operation metric.* Can be calculated on any non-abstract method.
+
+### Description
+
+Number of independent paths through a block of code \[[Lanza05](#Lanza05)\]. 
+Formally, given that the control flow graph of the block has n vertices, e edges and p connected components, the 
+Cyclomatic complexity of the block is given by `CYCLO = e - n + 2p` \[[McCabe76](#McCabe76)\]. In practice it can be 
+calculated by counting control flow statements following the standard rules given below.
+
+The standard version of the metric complies with McCabe's original definition:
+
+* Methods have a base complexity of 1.
+* +1 for every control flow statement (`if`, `case`, `catch`, `throw`, `do`, `while`, `for`, `break`, `continue`) and 
+conditional expression (`?:`) \[[Sonarqube](#Sonarqube)\]. Notice switch cases count as one, but not the switch itself: the point is that
+ a switch should have the same complexity value as the equivalent series of `if` statements.
+* `else`, `finally` and `default` don't count;
+* +1 for every boolean operator (`&&`, `||`) in the guard condition of a control flow statement. That's because
+Java has short-circuit evaluation semantics for boolean operators, which makes every boolean operator kind of a
+control flow statement in itself.
+ 
+### Versions
+
+* Version `CycloVersion#IGNORE_BOOLEAN_PATHS`: Boolean operators are not counted, nor are empty
```

Comment:
Right this can be more detailed. It was to provide results similar to what the old `StdCyclomaticComplexity` provided, but as the rule will be removed I guess it's not even for backwards compatibility